drivers/sensor/paw3212/paw3212.c

```diff
@@ -0,0 +1,524 @@
+	if (err) {
+		return err;
+	}
+
+	u8_t buf[] = {
+		SPI_WRITE_BIT | reg,
+		val
+	};
+	const struct spi_buf tx_buf = {
+		.buf = buf,
+		.len = ARRAY_SIZE(buf)
+	};
+	const struct spi_buf_set tx = {
+		.buffers = &tx_buf,
+		.count = 1
+	};
+
+	err = spi_write(dev_data->spi_dev, &spi_cfg, &tx);
+	if (err) {
+		LOG_ERR("Reg write failed on SPI write");
+		return err;
+	}
+
+	k_busy_wait(T_SCLK_NCS_WR);
+
+	err = spi_cs_ctrl(dev_data, false);
+	if (err) {
+		return err;
+	}
+
+	k_busy_wait(T_SWX);
+
+	return 0;
+}
+
+static void irq_handler(struct device *gpiob, struct gpio_callback *cb,
+			u32_t pins)
+{
+	sensor_trigger_handler_t handler;
+
+	k_spinlock_key_t key = k_spin_lock(&paw3212_data.lock);
+	handler = paw3212_data.data_ready_handler;
+	k_spin_unlock(&paw3212_data.lock, key);
+
+	struct sensor_trigger trig = {
+		.type = SENSOR_TRIG_DATA_READY,
+		.chan = SENSOR_CHAN_ALL,
+	};
+
+	handler(DEVICE_GET(paw3212), &trig);
+}
+
+static int paw3212_async_init_power_up(struct paw3212_data *dev_data)
+{
+	/* Reset sensor */
+
+	return reg_write(dev_data, PAW3212_REG_POWER_UP_RESET, 0x80);
+}
+
+static int paw3212_async_init_configure(struct paw3212_data *dev_data)
+{
+	int err;
+
+	u8_t product_id;
+	err = reg_read(dev_data, PAW3212_REG_PRODUCT_ID, &product_id);
+	if (err) {
+		LOG_ERR("Cannot obtain product id");
+		return err;
+	}
+
+	LOG_INF("Product ID: 0x%" PRIx8, product_id);
+	if (product_id != PAW3212_PRODUCT_ID) {
+		LOG_ERR("Invalid product ID (0x%" PRIx8")!", product_id);
+		return -EIO;
+	}
+
+	return err;
+}
+
+static void paw3212_async_init(struct k_work *work)
+{
+	struct paw3212_data *dev_data = &paw3212_data;
+
+	ARG_UNUSED(work);
+
+	LOG_DBG("PAW3212 async init step %d", dev_data->async_init_step);
+
+	dev_data->err = async_init_fn[dev_data->async_init_step](dev_data);
+	if (dev_data->err) {
+		LOG_ERR("PAW3212 initialization failed");
+	} else {
+		dev_data->async_init_step++;
+
+		if (dev_data->async_init_step == ASYNC_INIT_STEP_COUNT) {
+			dev_data->ready = true;
+			LOG_INF("PAW3212 initialized");
+		} else {
+			k_delayed_work_submit(&dev_data->init_work,
+					      async_init_delay[dev_data->async_init_step]);
+		}
+	}
+}
+
+static int paw3212_init_cs(struct paw3212_data *dev_data)
+{
+	int err;
+
+	dev_data->cs_gpio_dev =
+		device_get_binding(CONFIG_PAW3212_CS_GPIO_DEV_NAME);
+	if (!dev_data->cs_gpio_dev) {
+		LOG_ERR("Cannot get CS GPIO device");
+		return -ENXIO;
+	}
+
+	err = gpio_pin_configure(dev_data->cs_gpio_dev,
+				 CONFIG_PAW3212_CS_GPIO_PIN,
+				 GPIO_DIR_OUT);
+	if (!err) {
+		err = spi_cs_ctrl(dev_data, false);
+	} else {
+		LOG_ERR("Cannot configure CS PIN");
+	}
+
+	return err;
+}
+
+static int paw3212_init_irq(struct paw3212_data *dev_data)
+{
+	int err;
+
+	dev_data->irq_gpio_dev =
+		device_get_binding(CONFIG_PAW3212_IRQ_GPIO_DEV_NAME);
+	if (!dev_data->irq_gpio_dev) {
+		LOG_ERR("Cannot get IRQ GPIO device");
+		return -ENXIO;
+	}
+
+	err = gpio_pin_configure(dev_data->irq_gpio_dev,
```

Comment:
int err -= ...

Reply:
I would prefer to leave it as is - i.e. keep `int err` on top